src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -283,7 +286,10 @@ def infer_data_source_schema(
 
 def data_source_data_to_pandas_df(data: List[Any], schema: StructType) -> pd.DataFrame:
     columns = [col.name for col in schema.fields]
-    df = pd.DataFrame.from_records(data, columns=columns)
+    if not data:
+        return pd.DataFrame(columns=columns)
```

Comment:
is there a reason we did this special treatment? I tried `pd.DataFrame([],columns=["a", "b"])`, and it does not produce any error

Reply:
no special reason just my own code preference, I can consolidate the two